src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/hooks/useDefaultVMSource.ts

```diff
@@ -20,9 +21,20 @@ const useDefaultVMSource = (vm: V1VirtualMachine) => {
     defaultDiskSource.current = source;
   }, []);
 
+  const isDefaultDiskSource = () => {
+    const { storage: _, ...restDefaultSpec }: V1beta1DataVolumeSpec = {
+      ...(defaultDiskSource.current as V1beta1DataVolumeSpec),
+    };
+    const { storage: __, ...restCurrentSpec }: V1beta1DataVolumeSpec = {
+      ...(currentDiskSource as V1beta1DataVolumeSpec),
+    };
+
+    return isEqualObject(restDefaultSpec, restCurrentSpec);
+  };
+
   return {
     currentDiskSource,
-    isDefaultDiskSource: currentDiskSource === defaultDiskSource.current,
+    isDefaultDiskSource: isDefaultDiskSource(),
```

Comment:
why function if u simply return the result?

Reply:
No good reason, changed thanks :)